integrationTests/vm/txsFee/dynamicGasCost_test.go

```diff
@@ -0,0 +1,224 @@
+	trieKeysDepth := getTrieDepthForKeys(t, dataTrie, keys)
+
+	apiCallsCost := 3
+	loadValCost := 32
+	wasmOpsCost := 14
+
+	newContractCode := wasm.GetSCCode("../wasm/testdata/trieStoreAndLoad/storage.wasm")
+	latestGasSchedule := gasScheduleNotifier.LatestGasSchedule()
+	AoTPrepare := latestGasSchedule[common.BaseOperationCost]["AoTPreparePerByte"] * uint64(len(newContractCode)) / 2
+
+	gasCost := int64(apiCallsCost+loadValCost+wasmOpsCost) + int64(AoTPrepare)
+
+	for i, key := range keys {
+		trieLoadCost := getExpectedConsumedGasForTrieLoad(testContext, int64(trieKeysDepth[i]))
+
+		expectedGasCost := trieLoadCost + gasCost
+
+		fmt.Println("trie level", trieKeysDepth[i])
+		gasLimit = uint64(trieLoadCost) + 10000
+		testGasConsumedForDataTrieLoad(t, testContext, sndNonce, key, sndAddr, scAddress, gasPrice, gasLimit, expectedGasCost)
+		sndNonce++
+	}
+}
+
+func insertInDataTrie(
+	t *testing.T,
+	testContext *vm.VMTestContext,
+	nonce uint64,
+	maxTrieLevel int,
+	sndAddr []byte,
+	scAddress []byte,
+	gasPrice uint64,
+	gasLimit uint64,
+) [][]byte {
+
+	keys := integrationTests.GenerateTrieKeysForMaxLevel(maxTrieLevel, 2)
+	for _, key := range keys {
+		txData := []byte("trieStore@" + hex.EncodeToString(key) + "@" + hex.EncodeToString(key))
+		tx := vm.CreateTransaction(nonce, big.NewInt(0), sndAddr, scAddress, gasPrice, gasLimit, txData)
+		nonce++
+
+		returnCode, errProcess := testContext.TxProcessor.ProcessTransaction(tx)
+		require.Nil(t, errProcess)
+		require.Equal(t, vmcommon.Ok, returnCode)
+	}
+
+	return keys
+}
+
+func testGasConsumedForDataTrieLoad(
+	t *testing.T,
+	testContext *vm.VMTestContext,
+	nonce uint64,
+	key []byte,
+	sndAddr []byte,
+	scAddress []byte,
+	gasPrice uint64,
+	gasLimit uint64,
+	expectedGasConsumed int64,
+) {
+	testContext.CleanIntermediateTransactions(t)
+
+	txData := []byte("trieLoad@" + hex.EncodeToString(key) + "@00")
+	scr := &smartContractResult.SmartContractResult{
+		Nonce:    nonce,
+		Value:    big.NewInt(0),
+		RcvAddr:  scAddress,
+		SndAddr:  sndAddr,
+		Data:     txData,
+		GasLimit: gasLimit,
+		GasPrice: gasPrice,
+		CallType: 1,
+	}
+	returnCode, errProcess := testContext.ScProcessor.ProcessSmartContractResult(scr)
+	require.Nil(t, errProcess)
+	require.Equal(t, vmcommon.Ok, returnCode)
+
+	intermediate := testContext.GetIntermediateTransactions(t)
+	require.Equal(t, 1, len(intermediate))
+
+	gasConsumed := gasLimit - intermediate[0].GetGasLimit()
+	fmt.Println("gas consumed", gasConsumed)
+	fmt.Println("expected gas consumed", expectedGasConsumed)
+	require.Equal(t, uint64(expectedGasConsumed), gasConsumed)
+
+}
+
+func getTrieDepthForKeys(t *testing.T, tr common.Trie, keys [][]byte) []uint32 {
+	trieLevels := make([]uint32, len(keys))
+	for i, key := range keys {
+		_, depth, err := tr.Get(key)
+		require.Nil(t, err)
+		trieLevels[i] = depth
+	}
+
+	return trieLevels
+}
+
+func deployContract(t *testing.T, testContext *vm.VMTestContext, pathToContract string) []byte {
```

Comment:
I think we have this function already defined in the "utils" package

Reply:
Indeed. Used the func from utils.